File: bindings/lua/Polycode/GPUDrawOptions.lua
class "GPUDrawOptions"


function GPUDrawOptions:__getvar(name)
	if name == "depthTest" then
		return Polycode.GPUDrawOptions_get_depthTest(self.__ptr)
	elseif name == "depthWrite" then
		return Polycode.GPUDrawOptions_get_depthWrite(self.__ptr)
	elseif name == "linePointSize" then
		return Polycode.GPUDrawOptions_get_linePointSize(self.__ptr)
	elseif name == "alphaTest" then
		return Polycode.GPUDrawOptions_get_alphaTest(self.__ptr)
	elseif name == "backfaceCull" then
		return Polycode.GPUDrawOptions_get_backfaceCull(self.__ptr)
	elseif name == "depthOnly" then
		return Polycode.GPUDrawOptions_get_depthOnly(self.__ptr)
	elseif name == "forceMaterial" then
		return Polycode.GPUDrawOptions_get_forceMaterial(self.__ptr)
	elseif name == "blendingMode" then
		return Polycode.GPUDrawOptions_get_blendingMode(self.__ptr)
	elseif name == "scissorBox" then
		local retVal = Polycode.GPUDrawOptions_get_scissorBox(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Rectangle"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "enableScissor" then
		return Polycode.GPUDrawOptions_get_enableScissor(self.__ptr)
	elseif name == "drawColor" then
		local retVal = Polycode.GPUDrawOptions_get_drawColor(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Color"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	end
end

function GPUDrawOptions:__setvar(name,value)
	if name == "depthTest" then
		Polycode.GPUDrawOptions_set_depthTest(self.__ptr, value)
		return true
	elseif name == "depthWrite" then
		Polycode.GPUDrawOptions_set_depthWrite(self.__ptr, value)
		return true
	elseif name == "linePointSize" then
		Polycode.GPUDrawOptions_set_linePointSize(self.__ptr, value)
		return true
	elseif name == "alphaTest" then
		Polycode.GPUDrawOptions_set_alphaTest(self.__ptr, value)
		return true
	elseif name == "backfaceCull" then
		Polycode.GPUDrawOptions_set_backfaceCull(self.__ptr, value)
		return true
	elseif name == "depthOnly" then
		Polycode.GPUDrawOptions_set_depthOnly(self.__ptr, value)
		return true
	elseif name == "forceMaterial" then
		Polycode.GPUDrawOptions_set_forceMaterial(self.__ptr, value)
		return true
	elseif name == "blendingMode" then
		Polycode.GPUDrawOptions_set_blendingMode(self.__ptr, value)
		return true
	elseif name == "scissorBox" then
		Polycode.GPUDrawOptions_set_scissorBox(self.__ptr, value.__ptr)
		return true
	elseif name == "enableScissor" then
		Polycode.GPUDrawOptions_set_enableScissor(self.__ptr, value)
		return true
	elseif name == "drawColor" then
		Polycode.GPUDrawOptions_set_drawColor(self.__ptr, value.__ptr)
		return true
	end
	return false
end
function GPUDrawOptions:__delete()
	if self then Polycode.delete_GPUDrawOptions(self.__ptr) end
end

File: bindings/javascript/Polycode/GPUDrawOptions.js
function GPUDrawOptions() {
	Object.defineProperties(this, {
		'depthTest': { enumerable: true, configurable: true, get: GPUDrawOptions.prototype.__get_depthTest, set: GPUDrawOptions.prototype.__set_depthTest},
		'depthWrite': { enumerable: true, configurable: true, get: GPUDrawOptions.prototype.__get_depthWrite, set: GPUDrawOptions.prototype.__set_depthWrite},
		'linePointSize': { enumerable: true, configurable: true, get: GPUDrawOptions.prototype.__get_linePointSize, set: GPUDrawOptions.prototype.__set_linePointSize},
		'alphaTest': { enumerable: true, configurable: true, get: GPUDrawOptions.prototype.__get_alphaTest, set: GPUDrawOptions.prototype.__set_alphaTest},
		'backfaceCull': { enumerable: true, configurable: true, get: GPUDrawOptions.prototype.__get_backfaceCull, set: GPUDrawOptions.prototype.__set_backfaceCull},
		'depthOnly': { enumerable: true, configurable: true, get: GPUDrawOptions.prototype.__get_depthOnly, set: GPUDrawOptions.prototype.__set_depthOnly},
		'forceMaterial': { enumerable: true, configurable: true, get: GPUDrawOptions.prototype.__get_forceMaterial, set: GPUDrawOptions.prototype.__set_forceMaterial},
		'blendingMode': { enumerable: true, configurable: true, get: GPUDrawOptions.prototype.__get_blendingMode, set: GPUDrawOptions.prototype.__set_blendingMode},
		'scissorBox': { enumerable: true, configurable: true, get: GPUDrawOptions.prototype.__get_scissorBox, set: GPUDrawOptions.prototype.__set_scissorBox},
		'enableScissor': { enumerable: true, configurable: true, get: GPUDrawOptions.prototype.__get_enableScissor, set: GPUDrawOptions.prototype.__set_enableScissor},
		'drawColor': { enumerable: true, configurable: true, get: GPUDrawOptions.prototype.__get_drawColor, set: GPUDrawOptions.prototype.__set_drawColor}
	})
}
GPUDrawOptions.prototype.__get_depthTest = function() {
	return Polycode.GPUDrawOptions__get_depthTest(this.__ptr)
}

GPUDrawOptions.prototype.__set_depthTest = function(val) {
	Polycode.GPUDrawOptions__set_depthTest(this.__ptr, val)
}

GPUDrawOptions.prototype.__get_depthWrite = function() {
	return Polycode.GPUDrawOptions__get_depthWrite(this.__ptr)
}

GPUDrawOptions.prototype.__set_depthWrite = function(val) {
	Polycode.GPUDrawOptions__set_depthWrite(this.__ptr, val)
}

GPUDrawOptions.prototype.__get_linePointSize = function() {
	return Polycode.GPUDrawOptions__get_linePointSize(this.__ptr)
}

GPUDrawOptions.prototype.__set_linePointSize = function(val) {
	Polycode.GPUDrawOptions__set_linePointSize(this.__ptr, val)
}

GPUDrawOptions.prototype.__get_alphaTest = function() {
	return Polycode.GPUDrawOptions__get_alphaTest(this.__ptr)
}

GPUDrawOptions.prototype.__set_alphaTest = function(val) {
	Polycode.GPUDrawOptions__set_alphaTest(this.__ptr, val)
}

GPUDrawOptions.prototype.__get_backfaceCull = function() {
	return Polycode.GPUDrawOptions__get_backfaceCull(this.__ptr)
}

GPUDrawOptions.prototype.__set_backfaceCull = function(val) {
	Polycode.GPUDrawOptions__set_backfaceCull(this.__ptr, val)
}

GPUDrawOptions.prototype.__get_depthOnly = function() {
	return Polycode.GPUDrawOptions__get_depthOnly(this.__ptr)
}

GPUDrawOptions.prototype.__set_depthOnly = function(val) {
	Polycode.GPUDrawOptions__set_depthOnly(this.__ptr, val)
}

GPUDrawOptions.prototype.__get_forceMaterial = function() {
	return Polycode.GPUDrawOptions__get_forceMaterial(this.__ptr)
}

GPUDrawOptions.prototype.__set_forceMaterial = function(val) {
	Polycode.GPUDrawOptions__set_forceMaterial(this.__ptr, val)
}

GPUDrawOptions.prototype.__get_blendingMode = function() {
	return Polycode.GPUDrawOptions__get_blendingMode(this.__ptr)
}

GPUDrawOptions.prototype.__set_blendingMode = function(val) {
	Polycode.GPUDrawOptions__set_blendingMode(this.__ptr, val)
}

GPUDrawOptions.prototype.__get_scissorBox = function() {
	var retVal = new Rectangle()
	retVal.__ptr = 	Polycode.GPUDrawOptions__get_scissorBox(this.__ptr)
	return retVal
}

GPUDrawOptions.prototype.__set_scissorBox = function(val) {
	Polycode.GPUDrawOptions__set_scissorBox(this.__ptr, val.__ptr)
}

GPUDrawOptions.prototype.__get_enableScissor = function() {
	return Polycode.GPUDrawOptions__get_enableScissor(this.__ptr)
}

GPUDrawOptions.prototype.__set_enableScissor = function(val) {
	Polycode.GPUDrawOptions__set_enableScissor(this.__ptr, val)
}

GPUDrawOptions.prototype.__get_drawColor = function() {
	var retVal = new Color()
	retVal.__ptr = 	Polycode.GPUDrawOptions__get_drawColor(this.__ptr)
	return retVal
}

GPUDrawOptions.prototype.__set_drawColor = function(val) {
	Polycode.GPUDrawOptions__set_drawColor(this.__ptr, val.__ptr)
}

